src/SFA.DAS.LearnerDataMismatches.Domain/LearnerReport.cs

```diff
@@ -33,7 +30,15 @@ public LearnerReport(List<Payments.Model.Core.Entities.ApprenticeshipModel> appr
                 .Where(x => apps.Select(y => y.Ukprn).Contains(x.Ukprn))
                 .Select(x => new CollectionPeriod
             {
+                DataLocks = locks
+                    .Where(l => l.Ukprn == x.Ukprn && l.CollectionPeriod == x.CollectionPeriod)
+                    .SelectMany(l => l.NonPayablePeriods)
+                    .SelectMany(l => l.DataLockEventNonPayablePeriodFailures)
+                    .Select(l => (DataLock)l.DataLockFailure)
+                    .ToList(),
+                
                 Apprenticeship = apps.Find(a => a.Uln == x.LearnerUln && a.Ukprn == x.Ukprn),
```

Comment:
We may not need this if we expect only active apprenticeship record as the class parameter